DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/onearg.jl

```diff
@@ -0,0 +1,100 @@
+struct MooncakeOneArgPullbackPrep{Y,R} <: PullbackPrep
+    y_prototype::Y
+    rrule::R
+end
+
+function DI.prepare_pullback(
+    f, backend::AutoMooncake, x, ty::DI.Tangents, contexts::Vararg{Context,C}
+) where {C}
+    y = f(x, map(unwrap, contexts)...)
+    config = get_config(backend)
+    rrule = build_rrule(
+        get_interpreter(),
+        Tuple{typeof(f),typeof(x),typeof.(map(unwrap, contexts))...};
+        debug_mode=config.debug_mode,
+        silence_debug_messages=config.silence_debug_messages,
+    )
+    prep = MooncakeOneArgPullbackPrep(y, rrule)
+    DI.value_and_pullback(f, prep, backend, x, ty, contexts...)  # warm up
+    return prep
+end
+
+function DI.value_and_pullback(
+    f,
+    prep::MooncakeOneArgPullbackPrep{Y},
+    ::AutoMooncake,
+    x,
+    ty::DI.Tangents{1},
+    contexts::Vararg{Context,C},
+) where {Y,C}
+    dy = only(ty)
+    dy_righttype = convert(tangent_type(Y), dy)
+    new_y, (_, new_dx) = value_and_pullback!!(
```

Comment:
Will this work if there's a context argument? i.e. the second element of `Mooncake.value_and_pullback!!` is always a `Tuple` with as many elements as arguments to `f` + 1 (the +1 is because `f` itself gets a tangent). This looks to me like it only expects there to ever be two elements in the `Tuple`, but I'm assuming that there can be more in general.

Reply:
I was a bit surprised but this seems to work:
```julia
julia> (a, b) = (1, 2, 3)
(1, 2, 3)

julia> b
2
```